fv3gfs/util/communicator.py

```diff
@@ -398,43 +455,50 @@ def start_halo_update(self, quantity: Quantity, n_points: int) -> HaloUpdateRequ
         if n_points == 0:
             raise ValueError("cannot perform a halo update on zero halo points")
         tag = self._get_halo_tag()
-        send_requests = self._Isend_halos(quantity, n_points, tag=tag)
-        recv_requests = self._Irecv_halos(quantity, n_points, tag=tag)
-        return HaloUpdateRequest(send_requests, recv_requests)
-
-    def _Isend_halos(self, quantity: Quantity, n_points: int, tag: int = 0):
-        send_requests = []
+        recv_data = self._Irecv_halos(quantity, n_points, tag=tag)
+        send_data = self._Isend_halos(quantity, n_points, tag=tag)
+        return HaloUpdateRequest(send_data, recv_data, self.timer)
+
+    def _Isend_halos(
+        self, quantity: Quantity, n_points: int, tag: int = 0
+    ) -> HaloRequestSendList:
+        send_data = []
         for boundary in self.boundaries.values():
             with self.timer.clock("pack"):
-                data = boundary.send_view(quantity, n_points=n_points)
+                source_view = boundary.send_view(quantity, n_points=n_points)
                 # sending data across the boundary will rotate the data
                 # n_clockwise_rotations times, due to the difference in axis orientation.\
                 # Thus we rotate that number of times counterclockwise before sending,
                 # to get the right final orientation
-                data = rotate_scalar_data(
-                    data, quantity.dims, quantity.np, -boundary.n_clockwise_rotations
+                source_view = rotate_scalar_data(
+                    source_view,
+                    quantity.dims,
+                    quantity.np,
+                    -boundary.n_clockwise_rotations,
                 )
-            send_requests.append(
-                self._Isend(quantity.np, data, dest=boundary.to_rank, tag=tag)
+            send_data.append(
+                self._Isend(quantity.np, source_view[:], dest=boundary.to_rank, tag=tag)
```

Comment:
The `[:]` shouldn't be necessary, I think.
```suggestion
                self._Isend(quantity.np, source_view, dest=boundary.to_rank, tag=tag)
```

Reply:
Done